packages/react-tabs/src/components/Tab/useTabAnimatedIndicator.ts

```diff
@@ -0,0 +1,185 @@
+        `${pendingAnimationEasingTokens.declerateMax},` +
+        `${pendingAnimationEasingTokens.declerateMax}`,
+    },
+    '@media (prefers-reduced-motion: reduce)': {
+      ':after': {
+        transitionProperty: 'none',
+        transitionDuration: '0ms',
+      },
+    },
+  },
+  animatedVertical: {
+    ':after': {
+      transitionProperty: 'top, height, bottom',
+      transitionDuration:
+        `${pendingAnimationDurationTokens.slow},` +
+        `${pendingAnimationDurationTokens.slow},` +
+        `${pendingAnimationDurationTokens.slow}`,
+      transitionTimingFunction:
+        `${pendingAnimationEasingTokens.declerateMax},` +
+        `${pendingAnimationEasingTokens.declerateMax},` +
+        `${pendingAnimationEasingTokens.declerateMax}`,
+    },
+    '@media (prefers-reduced-motion: reduce)': {
+      ':after': {
+        transitionProperty: 'none',
+        transitionDuration: '0ms',
+      },
+    },
+  },
+  mediumHorizontal: {
+    ':after': {
+      left: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.mediumHorizontal})`,
+      right: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.mediumHorizontal})`,
+    },
+  },
+  mediumVertical: {
+    ':after': {
+      top: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.mediumVertical})`,
+      bottom: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.mediumVertical})`,
+    },
+  },
+  smallHorizontal: {
+    ':after': {
+      left: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.smallHorizontal})`,
+      right: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.smallHorizontal})`,
+    },
+  },
+  smallVertical: {
+    ':after': {
+      top: `calc(var(${indicatorCssVars.startOffsetVar}) + ${tabIndicatorPadding.smallVertical})`,
+      bottom: `calc(var(${indicatorCssVars.endOffsetVar}) + ${tabIndicatorPadding.smallVertical})`,
+    },
+  },
+});
+
+const calculateTabRect = (element: HTMLElement) => {
+  if (element) {
+    const parentRect = element.parentElement?.getBoundingClientRect() || { x: 0, y: 0, width: 0, height: 0 };
+    const tabRect = element.getBoundingClientRect();
+
+    return {
+      x: tabRect.x - parentRect.x,
+      y: tabRect.y - parentRect.y,
+      width: tabRect.width,
+      height: tabRect.height,
+    };
+  }
+  return undefined;
+};
+
+/**
+ * Adds additional styling to the active tab selection indicator to create a sliding animation.
+ */
+export const useTabAnimatedIndicatorStyles_unstable = (state: TabState): TabState => {
+  const activeIndicatorStyles = useActiveIndicatorStyles();
+
+  const { disabled, selectedValue, value, vertical } = state;
+  const selected = selectedValue === value;
+
+  const { previousSelectedValue, registeredTabs } = useContextSelector(TabListContext, ctx => ({
+    previousSelectedValue: ctx.previousSelectedValue,
+    registeredTabs: ctx.registeredTabs,
+  }));
```

Comment:
This probably is not the case (as `registeredTabs` will never change), but in general you don't want to combine selector functions.